receiver/prometheusreceiver/config.go

```diff
@@ -51,13 +53,40 @@ var _ config.Unmarshallable = (*Config)(nil)
 
 // Validate checks the receiver configuration is valid.
 func (cfg *Config) Validate() error {
-	if cfg.PrometheusConfig == nil {
+	promConfig := cfg.PrometheusConfig
+	if promConfig == nil {
 		return nil // noop receiver
 	}
-	if len(cfg.PrometheusConfig.ScrapeConfigs) == 0 {
+	if len(promConfig.ScrapeConfigs) == 0 {
 		return errors.New("no Prometheus scrape_configs")
 	}
 
+	// Reject features that Prometheus supports but that the receiver doesn't support:
+	// See:
+	// * https://github.com/open-telemetry/opentelemetry-collector/issues/3863
```

Comment:
nit: I'd prefer linking to docs, rather than issues from code comments.  Can we document what is supported/not supported in the README (and why), and link to that instead?

Reply:
@dashpole thank you for the review and for the comments! I've added what we don't support to the README. However, I think it is always useful to leave direct issue links in the code comments instead of people in the future having to hunt through and ask about various decisions -- this was actually a paint point for me when I had to resume work for the collector, I had to go look manually in Git blame history to see what or why things were done a certain way, given that I had no breadcrumbs left over.